docs/DeveloperGuide.md

```diff
@@ -217,7 +217,17 @@ _{more aspects and alternatives to be added}_
+
+#### Why it is implemented this way
+The `edit` command is implemented this way to ensure consistency with the other commands in the application. This helps to minimise any potential confusion for the users by standardising the prefixes that `edit` command takes in with the other relevant commands. In addition, it was intended for **EditCommandParser** to throw out a **ParseException** when none of the field changes an existing entry. This is to remind users in case they made a minor mistake, resulting in a supplier or warehouse to not update the way they intended for it to. Lastly, a command type prefix, `ct/COMMAND_TYPE` is required in the implementation of `edit` command to indicate whether user wishes to edit a warehouse or supplier entry. Without a command type prefix, an alternative would be for a `TYPE` parameter, where user have to indicate `supplier` or `warehouse`. However, this may not be suitable for our target user, who wishes to update stocks quickly. Hence, our team decided to use a command type prefix in place of a parameter. Furthermore, another alternative considered would be to create separate commands for warehouse and suppliers respectively. For example, `editw` and `edits` to represent edit warehouse and edit supplier. However, this might increase duplicated codes, since minimal changes to the code would be found for each class of command. By using a prefix, this helps us to reduce any potential code duplication. Therefore, our team decided to implement edit command by taking in prefixes and throwing our relevant exceptions at appropriate points after considering code quality and end user experience.
```

Comment:
I cant figure out why this paragraph is bold in the file. (Can click the three dots at the top right then click `view file`)

Minor issue:
Can make warehouse plural as well in this sentence - "Furthermore, another alternative considered would be to create separate commands for warehouse and suppliers respectively."

Reply:
I think I figured out why, there was not space between the last line of the paragraph and the line right after it (with -), will fix this in my next commit!

Will also fix the grammar as well, thanks for pointing that out.